static/src/javascripts/projects/commercial/modules/dfp/fill-advert-slots.ts

```diff
@@ -61,8 +61,17 @@ const fillAdvertSlots = async (): Promise<void> => {
 					desktop: [adSizes.billboard, createAdSize(900, 250)],
 				};
 			}
-			return new Advert(adSlot, additionalSizes);
-		});
+
+			try {
+				const advert = new Advert(adSlot, additionalSizes);
+				return advert;
+			} catch {
+				return null;
+			}
+		})
+		.filter((advert) => advert !== null)
+		.map((advert) => advert as Advert);
```

Comment:
Surely there's a better way to do this with Typescript?!

Reply:
You should be able to fix this with a [type guard](https://juhanajauhiainen.com/posts/how-to-make-typescript-understand-array-filter)!

```typescript
.filter((advert): advert is Advert => advert !== null)
```